apps/api/src/routes/application-router.ts

```diff
@@ -295,4 +296,66 @@ applicationRouter.post('/applications/reject', jsonParser, async (req, res) => {
 		});
 	}
 });
+
+applicationRouter.post('/applications/submit', jsonParser, async (req, res) => {
+	const { applicationId, role, signature } = req.body;
+
+	if (!applicationId || !role || !signature) {
+		res.status(400).json({ message: 'Missing required fields.' });
+	}
+
+	if (!applicationId || isNaN(parseInt(applicationId))) {
+		res.status(400).json({
+			message: 'Invalid request. ApplicationId is required and must be a valid number.',
+			errors: 'MissingOrInvalidParameters',
+		});
+	}
+
+	// Mock validation for now
+	if (signature !== 'mock_signature') {
+		res.status(400).json({ message: 'Invalid signature.' });
+	}
+
+	let newStatus = '';
+	if (role === 'APPLICANT') {
+		newStatus = 'Rep Review';
+	} else if (role === 'REP') {
+		newStatus = 'DAC Review';
+	} else {
```

Comment:
This `newStatus` variable is unused, this block should be changed to `if (!(role === 'APPLICANT' || role === 'REP'))`